src/controllers/generator.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import { ProblemException } from '../exceptions/problem.exception';
+
+import { Controller } from '../interfaces';
+
+import { generatorTemplateParametersValidationMiddleware } from "../middlewares/generator-template-parameters-validation.middleware";
+import { ArchiverService } from '../services/archiver.service';
+import { GeneratorService } from '../services/generator.service';
+import { prepareParserConfig } from '../utils/parser';
+
+/**
+ * Controller which exposes the Generator functionality
+ */
+export class GeneratorController implements Controller {
+  public path = '/generate';
+
+  private archiverService = new ArchiverService();
+  private generatorService = new GeneratorService();
+
+  private async generate(req: Request, res: Response) {
+    const zip = this.archiverService.createZip(res);
+    this.archiverService.appendHeaders(res);
+
+    let tmpDir: string;
+    try {
+      tmpDir = this.archiverService.createTempDirectory();
```

Comment:
I'm very concerned about the fact we are writing to disk here. We can easily run out of disk pretty fast.

Ideally, creating it in memory and streaming it directly via response will be cool. As well, concerns on memory usage are there but I think we should find the right tradeoff.

What do you think about this?

Reply:
I would say that we are sure, because final is always executed at the end after try or catch, and because of the fact that this operation is executed just after sending the response, the user does not wait for deleting the folder, so for me it is ok :)

Reply:
I think I answered you in another place: https://github.com/asyncapi/server-api/pull/7/files#r766057733